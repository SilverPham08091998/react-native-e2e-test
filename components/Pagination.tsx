import { useCallback } from 'react';
import { ScrollView } from 'react-native';
import Colors from 'constants/Colors';
import { View } from './View';
import { Text } from './Text';

type Item = object | number | string | boolean;
export type PaginationProps = {
  currentStep: number;
  data: Array<Item>;
};
export function Pagination(props: PaginationProps) {
  const { data, currentStep } = props;
  if (currentStep < 0) {
    throw Error('Invalid step index');
  }
  if (data.length > 5) {
    throw Error('Data length exceeded. Maximum 5 steps.');
  }
  const renderStep = useCallback(
    (_: Item, index: number) => {
      const isActivated = index + 1 === currentStep;
      const lightColor = isActivated ? Colors.light.tint : Colors.light.disabled;
      const darkColor = isActivated ? Colors.dark.tint : Colors.dark.disabled;
      return (
        <View
          key={`item-${index}`}
          style={{ height: 5, width: 30, borderRadius: 5 }}
          lightColor={lightColor}
          darkColor={darkColor}
        />
      );
    },
    [currentStep],
  );

  return (
    <View style={{ backgroundColor: 'transparent', alignItems: 'center' }}>
      <ScrollView
        horizontal
        contentContainerStyle={{ columnGap: 10 }}
        style={{ paddingVertical: 10 }}
      >
        {data.map(renderStep)}
      </ScrollView>
      <Text lightColor={Colors.light.textDisabled} darkColor={Colors.dark.textDisabled}>
        Step {currentStep}/{data.length}
      </Text>
    </View>
  );
}
