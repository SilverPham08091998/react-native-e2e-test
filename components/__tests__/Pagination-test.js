import * as React from 'react';
import renderer from 'react-test-renderer';

import { Pagination } from '../Pagination';

it(`renders correctly`, () => {
  const tree = renderer.create(<Pagination data={[1, 2, 3]} currentStep={1} />).toJSON();

  expect(tree).toMatchSnapshot();
});
