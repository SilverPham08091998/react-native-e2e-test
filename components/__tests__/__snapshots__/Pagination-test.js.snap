// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "#fff",
      },
      {
        "alignItems": "center",
        "backgroundColor": "transparent",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "backgroundColor": "#fff",
        },
        {
          "backgroundColor": "transparent",
          "columnGap": 10,
          "flexDirection": "row",
          "paddingVertical": 10,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#33449b",
          },
          {
            "borderRadius": 5,
            "height": 5,
            "width": 30,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "rgba(1,1,1,0.1)",
          },
          {
            "borderRadius": 5,
            "height": 5,
            "width": 30,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "backgroundColor": "rgba(1,1,1,0.1)",
          },
          {
            "borderRadius": 5,
            "height": 5,
            "width": 30,
          },
        ]
      }
    />
  </View>
  <Text
    style={
      [
        {
          "fontSize": 17,
          "fontWeight": "400",
          "letterSpacing": 0.3,
          "lineHeight": 20,
        },
        {
          "color": "#535353",
        },
        undefined,
      ]
    }
  >
    Step 
    1
    /
    3
  </Text>
</View>
`;
